src/shogun/converter/Isomap.h

```diff
@@ -24,14 +24,24 @@ namespace shogun
 class CFeatures;
 class CDistance;
 
-/** @brief class Isomap used to embed data using Isomap algorithm
- * as described in
+/** @brief The Isomap class is used to embed data using Isomap algorithm
+ * as described in:
  *
  * Silva, V. D., & Tenenbaum, J. B. (2003).
  * Global versus local methods in nonlinear dimensionality reduction.
  * Advances in Neural Information Processing Systems 15, 15(Figure 2), 721-728. MIT Press.
  * Retrieved from http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.9.3407&rep=rep1&type=pdf
  *
+ * The Isomap algorithm can be considered as a modification of the
+ * classic Multidimensional Scaling algorithm. The algorithm itself is
+ * consists following steps:
+ * - For each feature vector \f$x\in X\f$ find its \f$k\f$ nearest
+ * neighbours and construct the sparse neighbourhood graph.
+ * - Compute squared distances matrix \f$D\f$ such as \f$D\{i,j\} = d^2(x_i,x_j)\f$.
+ * - Relax distances with shortest(so-called geodesic) distances on the sparse neighbourhood graph (e.g. with sparse Dijkstra algorithm).
+ * - Center the matrix \f$D\f$ with subtracting row mean, column mean and adding to the grand mean. Multiply \f$D\f$ element-wise with \f$-0.5\f$.
+ * - Compute embedding with the \f$t\f$ eigenvectors that correspond to the largest eigenvalues of the matrix \f$D\f$; normalize these vectors with dividing each eigenvectors with square root of its corresponding eigenvalue. Form the final embedding with eigenvectors as rows and projected feature vectors as columns.
```

Comment:
Break the three lines above to have a similar line width as the one used in the previous paragraphs (we don't follow any specific line width, but it is nice to be consistent at least within files).